backend/controllers/iepHistory.js

```diff
@@ -0,0 +1,50 @@
+const moment = require('moment')
+
+const getIepHistoryFactory = elite2Api => {
+  const getIepHistory = async (context, offenderNo) => {
+    const bookingDetails = await elite2Api.getDetails(context, offenderNo)
+    const iepSummary = await elite2Api.getIepSummaryWithDetails(context, bookingDetails.bookingId)
+    const iepHistoryDetails = []
+
+    /* eslint-disable no-restricted-syntax */
+    for (const details of iepSummary.iepDetails) {
+      /* eslint-disable no-await-in-loop */
+      const iepEstablishment = await elite2Api.getAgencyDetails(context, details.agencyId)
```

Comment:
I would have thought that an offender is likely to have many ieps at the same agency? - you could probably significantly reduce the number of api calls here by getting a list of the distinct agencyIds and then doing the call once for each agency id?

Reply:
You are very right! I have created unique sets of the values and only issue the call once per value.